False positive deprecated-method with alias to non-deprecated method

### Bug description

With pylint 2.12.2 and python 3.8 the following code:

```python
import time

def test(clock = time.time):
    clock()
```

Produces a false-positive `deprecated-method` warning:

```
test.py:4:4: W1505: Using deprecated method clock() (deprecated-method)
```

I think this is related to the deprecated alias detection added in https://github.com/PyCQA/pylint/issues/4618 as the issue does not reproduce with that change reverted.

### Configuration

_No response_

### Command used

```shell
pylint test.py
```


### Pylint output

```shell
test.py:4:4: W1505: Using deprecated method clock() (deprecated-method)
```


### Expected behavior

No warning.

### Pylint version

```shell
2.12.2
```


### OS / Environment

Ubuntu 18.04

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/5886.false_positive
+++ b/doc/whatsnew/fragments/5886.false_positive
@@ -1,0 +1,3 @@
+Fix ``deprecated-method`` false positive when alias for method is similar to name of deprecated method.
+
+Closes #5886

--- a/pylint/checkers/deprecated.py
+++ b/pylint/checkers/deprecated.py
@@ -213,16 +213,7 @@ class DeprecatedMixin(BaseChecker):
             # Not interested in other nodes.
             return
 
-        if hasattr(inferred.parent, "qname") and inferred.parent.qname():
-            # Handling the situation when deprecated function is
-            # alias to existing function.
-            qnames = {
-                inferred.qname(),
-                f"{inferred.parent.qname()}.{func_name}",
-                func_name,
-            }
-        else:
-            qnames = {inferred.qname(), func_name}
+        qnames = {inferred.qname(), func_name}
         if any(name in self.deprecated_methods() for name in qnames):
             self.add_message("deprecated-method", node=node, args=(func_name,))
             return

--- a/tests/checkers/unittest_deprecated.py
+++ b/tests/checkers/unittest_deprecated.py
@@ -108,23 +108,22 @@ class TestDeprecatedChecker(CheckerTestCase):
 
     def test_deprecated_method_alias(self) -> None:
         # Tests detecting deprecated method defined as alias
-        # to existing method
         node = astroid.extract_node(
             """
         class Deprecated:
-            def _deprecated_method(self):
+            def deprecated_method(self):
                 pass
 
-            deprecated_method = _deprecated_method
+            new_name = deprecated_method
 
         d = Deprecated()
-        d.deprecated_method()
+        d.new_name()
         """
         )
         with self.assertAddsMessages(
             MessageTest(
                 msg_id="deprecated-method",
-                args=("deprecated_method",),
+                args=("new_name",),
                 node=node,
                 confidence=UNDEFINED,
                 line=9,
@@ -135,6 +134,22 @@ class TestDeprecatedChecker(CheckerTestCase):
         ):
             self.checker.visit_call(node)
 
+    def test_not_deprecated(self) -> None:
+        # Tests detecting method is NOT deprecated when alias name is a deprecated name
+        node = astroid.extract_node(
+            """
+        class Deprecated:
+            def not_deprecated(self):
+                pass
+
+            deprecated_method = not_deprecated
+
+        d = Deprecated()
+        d.deprecated_method()
+        """
+        )
+        with self.assertNoMessages():
+            self.checker.visit_call(node)
     def test_no_message(self) -> None:
         # Tests not raising error when no deprecated functions/methods are present.
         node = astroid.extract_node(

--- a/tests/checkers/unittest_deprecated.py
+++ b/tests/checkers/unittest_deprecated.py
@@ -150,6 +150,7 @@ class TestDeprecatedChecker(CheckerTestCase):
         )
         with self.assertNoMessages():
             self.checker.visit_call(node)
+
     def test_no_message(self) -> None:
         # Tests not raising error when no deprecated functions/methods are present.
         node = astroid.extract_node(

--- a/tests/functional/d/deprecated/deprecated_methods_py38.py
+++ b/tests/functional/d/deprecated/deprecated_methods_py38.py
@@ -4,9 +4,9 @@ import base64
 import inspect
 import logging
 import nntplib
+import time
 import unittest
 import xml.etree.ElementTree
-
 
 class MyTest(unittest.TestCase):
     def test(self):
@@ -51,3 +51,7 @@ class Deprecated:  # pylint: disable=too-few-public-methods
 
 d = Deprecated()
 d.deprecated_method()  # [deprecated-method]
+
+def test(clock = time.time):
+    """time.clock is deprecated but time.time via an alias is not!"""
+    clock()

